src/axolotl/utils/data.py

```diff
@@ -242,14 +242,16 @@ def for_d_in_datasets(dataset_configs):
             local_path = Path(config_dataset.path)
             if local_path.exists():
                 if local_path.is_dir():
-                    # TODO dirs with arrow or parquet files could be loaded with `load_from_disk`
-                    ds = load_dataset(
-                        config_dataset.path,
-                        name=config_dataset.name,
-                        data_files=config_dataset.data_files,
-                        streaming=False,
-                        split=None,
-                    )
+                    if any(local_path.glob("*.arrow")) or any(local_path.glob("*.parquet")):
+                        ds = load_from_disk(config_dataset.path)
```

Comment:
@winglian , this should solve the issue with changing to `load_from_disk` right?